### PR TITLE
feat: add favorites management (fixes #111)

### LINKED ISSUE #111: Add favorites management
**Is your feature request related to a problem? Please describe.**

Users should be able to add a resource to a locally-stored favorites list, and view and manage these favorites from a dedicated page.https://github.com/platform-coop-toolkit/coop-library/issues/new/choose

**Describe the solution you'd like**

See wires in [Figma](https://www.figma.com/file/aqvUsF1E4um0qcYC3yF3FXD6/Resource-Library-Wireframes-MB?node-id=410%3A4908) and XD, as well as reference implementation of local storage in pwa-experiments.netlify.com.

**Describe alternatives you've considered**

Not applicable.

**Additional context**

Not applicable.

### PR DESCRIPTION
**Pull request by greatislander**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

Adds client- and server-side favorites management.

**Completed:**

- [x] Add to user favorites via single resource view (requires JS).
- [x] Increment resource favorite count via single resource view (requires JS).
- [x] Remove from user favorites via single resource view (requires JS).
- [x] Decrement resource favorite count via single resource view (requires JS).
- [x] Switch button icon to solid when favorited (requires https://github.com/platform-coop-toolkit/pinecone/pull/194).
- [x] View user favorites list.
- [x] Allow user to remove resources directly from their favorites list (requires JS).
- [x] Reflect favorite status of resources on cards in all views.
- [x] Notify user of success or failure when adding/removing favorite.

## Steps to test

1. Visit a single resource.
2. Click the "Add to favorites" button.
3. Navigate to 'My resources &rarr; Favorites'.
4. Click the "Remove" button.

**Expected behavior:**

- Button switches to "Remove from favorites".
- Cookie value for the "favorites" key is updated with the current resource ID added to the comma-separated list of resource IDs.
- Viewing the resource API data at `/wp-json/wp/v2/resources/<ID>/` shows that the value for `lc_resource_favorites` in the `meta` object has incremented by one.
- Navigating to 'My resources &rarr; Favorites' shows the resource in the displayed list of favorites.
- Clicking the "Remove" button below the resource removes it from the list.
- Users are notified of success or failure of all favorite/unfavorite operations.

## Additional information

Styling of the favorites page will be addressed once https://github.com/platform-coop-toolkit/pinecone/issues/205 is closed.

## Related issues

- Fixes #111.

### CODE REVIEW

**Comment by greatislander** [comment]
@jhung I'd love to get your feedback on the functionality as it exists. You'll want to pull the latest changes from the staging site to test it. If there's any underlying issues I'd like to address them before tackling favorites management on the favorites page and adding notifications.

**Comment by greatislander** [comment]
@cherylhjli Tagging you in on this as I will need the completed designs for the favorites page to finish this feature. But this isn't a blocker as I have other things to work on in the meantime.

**Comment by jhung** [comment]
Existing notification created by the same trigger should be removed before a new one is displayed.

Otherwise you can have a situation like this which is confusing:
image

For a fun effect:
1. resize the window narrow
2. view a resource like "_Trust and the Sharing Economy | Data Dialogs 2016_".
3. Select Add and remove favourites couple times.
4. Then scroll up and down... and watch the notifications stack in a satisfying way. 👍

**Comment by greatislander** [comment]
@cherylhjli I'd like to test this with you tomorrow via screen sharing.

File: resources/assets/scripts/util/addNotification.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    case 'error':
+      icon = '<svg class="icon icon--error" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><g id="error" fill="currentColor"><path d="m10.09 1a9 9 0 1 0 9 9 9 9 0 0 0 -9-9zm0 16a7 7 0 1 1 7-7 7 7 0 0 1 -7 7z"/><path d="m12.85 7.18a1 1 0 0 0 -1.41 0l-1.35 1.35-1.35-1.35a1 1 0 0 0 -1.41 0 1 1 0 0 0 0 1.42l1.35 1.4-1.35 1.29a1 1 0 0 0 1.41 1.42l1.35-1.35 1.35 1.35a1 1 0 0 0 1.41-1.42l-1.35-1.29 1.35-1.4a1 1 0 0 0 0-1.42z"/></g></svg>';
+      break;
+    case 'info':
+    default:
+      icon = '<svg class="icon icon--info" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><g id="info" fill="currentColor"><path d="m10 19.05a9.05 9.05 0 1 0 -9-9 9.06 9.06 0 0 0 9 9zm0-16.05a7.05 7.05 0 1 1 -7 7 7.05 7.05 0 0 1 7-7z"/><path d="m9.91 14.41a1 1 0 0 0 1-1v-4a1 1 0 0 0 -2 0v4a1 1 0 0 0 1 1z"/><path d="m9.89 7.4a1 1 0 1 0 -1-1 1 1 0 0 0 1 1z"/></g></svg>';
+  }
+
+  const alert = `
+    <div class="notification notification--${type}">
```
Comment (jhung) [inline resources/assets/scripts/util/addNotification.js]: `role=alert` is missing from the notification. No longer being announced by NVDA.

Comment (greatislander) [reply resources/assets/scripts/util/addNotification.js]: @jhung fixed now!

**Pull request merged**